File: download_rtc.py
#!/usr/bin/python

import serial, time, sys, fileinput
 
#open and configure serial port
ser = serial.Serial(
   # port='/dev/ttyAMA0',
	port='/dev/ttyUSB0',
    baudrate=19200,
    parity=serial.PARITY_NONE,
    stopbits=serial.STOPBITS_ONE,
    bytesize=serial.EIGHTBITS,
    timeout = .1	
)

#first, clear out 210 buffer
count = 0
while (count < 3):
 count +=1  
 ser.write("\r") 
 time.sleep(.1)
 

#open file for download
file = open("downloadrtc.mem", "w+")


#send command to rc210 to start download
ser.write("1SendRTCEram\r\n")

indata ="" 
Counter = 0
progresscounter = 0
var = 1

while var==1:
  indata = ser.readline()
  indata = indata.strip()
#  print indata  
  
  
  print "INDATA: ", indata
  if (indata=="T3Complete"): 
    file.write(indata)
    break
  
  if(indata[0:2] =="-R"):	
     Counter = Counter + 1
 #    break	
	 
  if(indata !=""):
    Counter = 0
  indata = indata + "\r\n"
  if(indata[0] =="T"): file.write(indata)
  time.sleep(.1)
  ser.write("r\r\n")  
  sys.stdout.flush() 
  
  if Counter > 10:
    file.close()
    sys.exit("RC210 did not respond. Exiting")
	
print "\nDownload Complete"
 
file.close()
sys.exit()

File: downloadrtc.php
<?php

$port = "/dev/ttyUSB0";
$cmd = "1SendRTCEram";
$eRam = "";
$memfile = fopen("downloadrtc.mem", "w\n");

$c = exec('stty -F '.$port.' cs8 -parenb -cstopb -echo raw speed 19200');

if(!file_exists($port)) {
	echo "Device doesn't exist.";
	die();
} else {

	$f = fopen($port, "w+");
	fwrite($f, $cmd."\r");

sleep(0.5);
$indata="";
$timediff =0;
$starttime = date('s');
$substring = "";
$v = 1;

while ($v == 1){	
	$indata = fgets($f);	
		
	if($indata != ""){
		echo "INDATA " . "$indata\r";	
		fwrite($memfile, $indata);	#write to download.mem
		fwrite($f,"r\r\n");	
		$starttime = time(); //reset timer		
	}		
	
		
	if($indata == "TimeOut"){die("Timeout from controller");}	
	
	if($indata == "T3Complete"){
		fclose($memfile);
		$indata = fgets($f); //read final +SENDE and clear from buffer
		exit();
	}	

	$substring = substr($indata,0,1);		
	if($substring =="-"){    		
		print "We encountered an error when receiving $indata". "Please close window and try again\r\n";	
		fclose($memfile);
		exit();		
	}
	
	$endtime = date('s');		
	$timediff = $endtime - $starttime;
	
	if($timediff > '4'){
		print"Serial Timeout! There was no response from the RC210. Please close this window, check your serial connections and try again\r\n";			
		fclose($memfile);
		exit();
	}
	$indata ="";

}//while
}

?>